__tests__/app.test.js

```diff
@@ -98,9 +98,92 @@ describe("GET /api/articles/:article_id", () => {
       .get("/api/articles/999")
       .expect(404)
       .then(({ body }) => {
-
         expect(body.msg).toBe("No Article With That ID");
       });
   });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("responds with status 200", () => {
+    const patch = {
+      inc_votes: 1,
+    };
+
+    return request(app).patch("/api/articles/2").send(patch).expect(200);
+  });
+  test("responds with updated object with key describing the data and the updated object article its value", () => {
+    const patch = {
+      inc_votes: 1,
+    };
+
+    return request(app)
+      .patch("/api/articles/2")
+      .send(patch)
+      .then(({ body }) => {
+        expect(body).toBeInstanceOf(Object);
+        expect(body.updatedArticle).toBeInstanceOf(Object);
+      });
+  });
+  test("the updated object contains the correct keys and information type, with the votes incremented by the request body", () => {
+    const patch = {
+      inc_votes: 1,
+    };
 
+    return request(app)
+      .patch("/api/articles/2")
+      .send(patch)
+      .then(({ body }) => {
+        expect(body.updatedArticle).toEqual(
+          expect.objectContaining({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: 1,
+          })
+        );
+      });
+  });
+  test("votes is decremented by the request body", () => {
+    const patch = {
+      inc_votes: -100,
+    };
+
+    return request(app)
+      .patch("/api/articles/2")
+      .send(patch)
+      .then(({ body }) => {
+        expect(body.updatedArticle).toEqual(
+          expect.objectContaining({
+            votes: -100,
+          })
+        );
+      });
+  });
+  test("status: 400 and 'Bad Request' for malformed request body", () => {
+    const patch = {};
+
+    return request(app)
+      .patch("/api/articles/2")
+      .send(patch)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("status: 400 and 'Bad Request' for incorrect value type", () => {
+    const patch = {
+      inc_votes: "notNumber",
+    };
+
+    return request(app)
+      .patch("/api/articles/2")
+      .send(patch)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
```

Comment:
Would add a test in here to check we send a bad request response when you send a patch to a invalid ID (e.g. article id = 'banana')

Also a test to see the right response when a article ID that isn't found (e.g. article Id = 4000) is used in a patch request